src/vaults/VaultOps.ts

```diff
@@ -198,6 +206,36 @@ async function mkdir(
   });
 }
 
+/**
+ * Adds an empty directory to the root of the vault.
+ * i.e. mkdir("folder", { recursive = false }) creates the "<vaultDir>/folder" directory
+ */
+// TODO: add log messages back
+async function* makeDirectories(
+  vault: Vault,
+  dirPaths: Array<string>,
+  fileOptions?: FileOptions,
+  logger?: Logger,
+): AsyncGenerator<SuccessMessage> {
+  const recursive = fileOptions?.recursive ?? false;
+  yield* vault.writeG(async function* (efs): AsyncGenerator<SuccessMessage> {
+    for (const dirPath of dirPaths) {
+      try {
+        await efs.mkdir(dirPath, fileOptions);
+        yield { success: true };
+      } catch (e) {
+        if (e.code === 'ENOENT' && !recursive) {
+          yield { success: false, error: { code: 'ENOENT', path: dirPath } };
+        }
+        if (e.code === 'EEXIST') {
+          yield { success: false, error: { code: 'EEXIST', path: dirPath } };
+        }
+        throw e;
```

Comment:
I'm not sure how to handle catching and throwing errors without exiting the loop. So far, I've tried creating a `vaultsErrors` error object like how `mkdir` does, but that didn't work and I was unable to deserialise the object back into an actual error. It remained as a serialised JSON object.

I've tried returning error strings directly, formatted exactly as they should be displayed on the client side, but this seems unoptimal, so I didn't go with that.

I need some help in figuring how to properly pass errors back to the caller without returning or throwing anything. @tegefaulkes

Reply:
The command `secrets touch` will have the same issues, as it also needs to do this. Previously, I literally returned the error we need to print, like in my second suggested method. After a solution is found, I will update the relevant handler (`VaultsSecretsGet`) to instead return properly serialised errors and not just the actual error strings. The error strings need to be the responsibility of the client, not the handler itself.

Reply:
After discussing this with Brian, the conclusion we have arrived at is to keep it as-is. Don't change the behaviour of returning a hard-coded string for `VaultsSecretGet` and implement the same for `VaultsSecretsMkdir`. We can update it later as needed, but we need to keep it simple and roll out features quickly.